### PR TITLE
F/275

### PR DESCRIPTION
**Pull request by icarus-sparry**
2 progress meter related changes, as 3 patches.

1) Fix the fix-missing-file so it doesn't get upset with extra dots coming out partway through other output. This will happen when we flush the progress meter.

2) Make the progress meter more accurate, and show 100% at the end of each step.  Issue #275 

3) Flush the non interactive progress meter, Issue #256 - adding this breaks the fix-missing-file.

### CODE REVIEW

**Comment by matthewrsj** [comment]
Please rebase to latest master to fix CI error.

File: src/helpers.c [hunk 1]
```diff
@@ -40,7 +40,6 @@
 #include "signature.h"
 #include "swupd.h"
 
-static int last_percentage = -1;
```
Comment (matthewrsj) [inline src/helpers.c]: `make compliant` complains about an extra line here.

**Comment by matthewrsj** [comment]
+1, I'll merge this after the code style fix is made.

**Comment by icarus-sparry** [comment]
Style updated. It does strike me that we should be sending progress meters to stderr (i.e. the terminal) rather than stdout to maybe be consumed by another program.

**Comment by matthewrsj** [comment]
I agree. #294 open.

**Pull request merged**